packages/app-desktop/gui/MenuBar.tsx

```diff
@@ -308,24 +309,17 @@ function useMenu(props: Props) {
 
 		if (Array.isArray(path)) path = path[0];
 
-		const modalMessage = _('Importing from "%s" as "%s" format. Please wait...', path, module.format);
-
-		void CommandService.instance().execute('showModalMessage', modalMessage);
-
 		// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
 		const errors: any[] = [];
 
+		const taskId = `interop-import-${path}}`;
 		const importOptions = {
 			path,
 			format: module.format,
 			outputFormat: module.outputFormat,
 			// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
-			onProgress: (status: any) => {
-				const statusStrings: string[] = Object.keys(status).map((key: string) => {
-					return `${key}: ${status[key]}`;
-				});
-
-				void CommandService.instance().execute('showModalMessage', `${modalMessage}\n\n${statusStrings.join('\n')}`);
+			onProgress: (_status: any) => {
+				TaskProgressUIService.onTaskProgress(taskId, _status);
```

Comment:
Looking at the [Evernote importer](https://github.com/laurent22/joplin/blob/9eb4944614ff00aa6e047803395c1c276d70b56e/packages/lib/import-enex.ts#L505), it seems that 1) `status` is an object with a `created` property 2) the total number of notes to import is unknown until import completes.

For now, it might be better to show indeterminate progress for import tasks. Supporting `onProgress` would still be useful for [exporting (example)](https://github.com/laurent22/joplin/blob/9eb4944614ff00aa6e047803395c1c276d70b56e/packages/app-mobile/components/screens/ConfigScreen/NoteExportSection/NoteExportButton.tsx#L55) and #10527.